### PR TITLE
UI Refactor

### PR DESCRIPTION
**Pull request by Loerad**
Hello

I have refactored most of the UI to the new system. For all new UI we should use the UI Toolkit, which if you need help just ask me.

I would like some feedback on font sizes and the color of text as I am unsure if I am happy with them, and of course any issues you come across.

The elements refactored are the HUD while playing the game, the main menu, the item panel, tutorial and game over screen. I will not be refactoring the UI for the high score system as we may be refactoring that whole system later on and can be done then.

### CODE REVIEW

**Review by kblack89 — CHANGES_REQUESTED** [review]
There is an issue where you cannot submit a name for high score and also main menu button stops working in the score screen

**Review by AlexReidNZ — APPROVED** [review]
The new UI using the UI toolkit looks and functions great.

File: Assets/Scripts/GameManager.cs [hunk 1]
```diff
@@ -22,25 +33,25 @@ public void GameOver()
...
         Timer.CullEnemies();
-   }
+    }
 
-    public void Restart() 
+    private void Restart(ClickEvent click)
     {
         ResetVariables();
-        SceneManager.LoadScene("MainScene");
+        SceneManager.LoadScene("UIRemaster");
```
[the remark this answers is not in the document]
Comment (Loerad) [reply Assets/Scripts/GameManager.cs:45]: This will happen in staging

**Comment by Loerad** [comment]
> There is an issue where you cannot submit a name for high score 

This issue is something that is not being solved in this branch

> main menu button stops working in the score screen

This has been resolved

**Review by PBWiseman — CHANGES_REQUESTED** [review]
New UI looks great! Well done learning and implementing the new system. The game over screen is currently set to visible by default so it shows up when loading the level. Once that is fixed, I am happy for this to be merged

**Review by PBWiseman — APPROVED** [review]

**Review by kblack89 — APPROVED** [review]
All previous issues have been resolved so this is good to go and the new code make sense and I cant see any apparent issues

**Pull request merged**